### ISSUE TITLE
 Pubsub emulator - Subscriptions seek to a point in time does not replay messages

### ISSUE THREAD

**Issue by maczech** [googleapis/java-pubsub#584]
I'm trying to replay all the messages for the subscription.
The same code works with GCloud but it doesn't with PubSub emulator.
For PubSub emulator SeekResponse is empty( according to the documentation this means that call was successful) but I'm not receiving old messages. 

#### Steps to reproduce

  1. Start pubsub as a docker container with entrypoint: 
  ```bash
    gcloud beta emulators pubsub start --project test-project --host-port 0.0.0.0:8085
  ```
Image: gcr.io/google.com/cloudsdktool/cloud-sdk:334.0.0-emulators(tested with previous versions)
  
  2. Start application and generate some messages.
  3. Use seek to replay all messages from the subscription

#### Code example

```java
  
  @Override
  protected SubscriptionAdminSettings createSubscriptionAdminSettings() throws IOException {
        ManagedChannel channel =
        ManagedChannelBuilder.forTarget(environmentChecker.getLocalHostAndPort().get())
            .usePlaintext()
            .build();
    TransportChannelProvider channelProvider =
        FixedTransportChannelProvider.create(GrpcTransportChannel.create(channel));
    return SubscriptionAdminSettings.newBuilder()
        .setTransportChannelProvider(channelProvider)
        .setCredentialsProvider( FixedCredentialsProvider.create(NoCredentialsProvider.create().getCredentials()))
        .build();
  }


public void replayMessages(String subscriptionName) {
    try (SubscriptionAdminClient subscriptionAdminClient =
        SubscriptionAdminClient.create(createSubscriptionAdminSettings())) {
      Duration messageRetentionDuration =
          subscriptionAdminClient
              .getSubscription(
                  GetSubscriptionRequest.newBuilder().setSubscription(subscriptionName).build())
              .getMessageRetentionDuration();
      LocalDateTime retentionTime =
          LocalDateTime.now().minusSeconds(messageRetentionDuration.getSeconds());
      Timestamp retentionTimeEpoch =
          Timestamp.newBuilder()
              .setSeconds(retentionTime.atZone(ZoneOffset.UTC).toEpochSecond())
              .build();

      SeekRequest request =
          SeekRequest.newBuilder()
              .setSubscription(subscriptionName)
              .setTime(retentionTimeEpoch)
              .build();
      subscriptionAdminClient.seek(request);
    } catch (IOException e) {
      logger.atSevere().withCause(e).log("Cannot replay messages");
    }
  }
```

**Comment by anguillanneuf** [comment]
@maczech 👋 thank you for reporting this issue!

This is indeed a bug within the Pub/Sub emulator. It does not retain messages when retention is set to `true` during subscription creation. It only does so if the subscription is explicitly updated with retention. We are cutting a new release for the emulator and will keep you posted. 

In the meanwhile, the following workaround allows you to replay messages from the time the `UpdateSubscriptionRequest` is made:

```java
// 1. Create a pull subscription.

// 2. Update it to retain acked messages

     // String projectId = "abc";
     // String topicId = "may";
     // String subscriptionId = "six";

      TopicName topicName = TopicName.of(projectId, topicId);
      ProjectSubscriptionName subscriptionName = ProjectSubscriptionName.of(projectId, subscriptionId);

      Subscription subscription = Subscription.newBuilder()
          .setTopic(topicName.toString())
          .setName(subscriptionName.toString())
          .setRetainAckedMessages(true)
          .build();

      FieldMask updateMask = FieldMask.newBuilder().addPaths("retain_acked_messages").build();

      try {
        UpdateSubscriptionRequest request =
            UpdateSubscriptionRequest.newBuilder()
                .setSubscription(subscription)
                .setUpdateMask(updateMask)
                .build();
        Subscription response = subscriptionAdminClient.updateSubscription(request);
        System.out.println("Updated pull subscription: " + response);
      } catch (StatusRuntimeException e) {
        // ...
      }

// 3. Seek to the subscription creation time

// 4. Receive messages since creation over and over again
```

**Comment by anguillanneuf** [comment]
@maczech This issue is resolved in Cloud SDK `341.0.0` and later. 

I have uploaded my own testing code. Feel free to follow the steps if you also want to verify that it indeed works: https://github.com/anguillanneuf/PubSubEmulatorExamples#replaying-messages
